test/Main.hs

```diff
@@ -100,18 +95,23 @@ parseResponses :: TestTree
       , (JSON.object [("foo", JSON.object empty)], "Invalid status code pattern: \"foo\"")
       ]
 
-    mkAccepted :: (JSON.Value, Paths.Responses) -> TestTree
-    mkAccepted (input, expected)
-      = testCase (convertText ("accepts: " <> show input))
-      $ pure expected @=? parseJSON input
-
-    mkRejected :: (JSON.Value, Text) -> TestTree
-    mkRejected (input, expectedMessage)
-      = testCase (convertText ("rejects: " <> show input))
-      $ Left ("Error in $: " <> convertText expectedMessage) @=? parseJSON input
-
-    parseJSON :: JSON.Value -> Either String Paths.Responses
-    parseJSON = JSON.parseEither JSON.parseJSON
-
 main :: IO ()
 main = defaultMain suite
+
+mkAccepted
+  :: (Eq a, JSON.FromJSON a, Show a)
+  => (JSON.Value, a)
+  -> TestTree
+mkAccepted (input, expected)
+  = testCase (convertText ("accepts: " <> show input))
```

Comment:
Can this be written as `testCase $ convertText $ "accepts: " <> show input`?

Reply:
The next line is still an argument, and it contains a binary operator that would inference with that. I could remove the inner parens.